kick_fifo.asm

```diff
@@ -2,49 +2,75 @@
 
 bits 32
 
+%define CACHE_PUSH_MASTER_STATE     0xFD003200
+%define CACHE_PUSH_STATE            0xFD003220
+%define DMA_PUSH_ADDR               0xFD003240
+%define DMA_PULL_ADDR               0xFD003244
+
 kick_fifo:
 
-; Load expected_put into reg
-mov edx, dword [esp+4]
+push ebp
+mov ebp, esp
+sub esp, 4
```

Comment:
What's the purpose of decrementing the stack pointer here? My x86 assembly is a bit rusty, so I'm probably just missing something

Reply:
Cruft from my flailing around trying to get this to work on xemu as well as HW where I ran out of registers; removed